### ISSUE TITLE
Setup guide side panel open by default

### ISSUE THREAD

**Issue by tealjulia** [airbytehq/airbyte#12731]
## Tell us about the problem you're trying to solve
In https://github.com/airbytehq/airbyte/issues/10644 we added a documentation side panel that opens in place of the popover modal we had previously.  This allows a user to interact with documentation while creating their source.

This should be open by default so users are more strongly encouraged to use the setup guide while completing the form.

![Screen Shot 2022-05-10 at 8 54 04 AM](https://user-images.githubusercontent.com/63751206/167633024-3b73f462-7e6a-4419-b99c-df365fc9dd03.png)

## Describe the solution you’d like

Before we begin this work, there are some things to consider, plan, and design (PM + design input needed):
- What should a user see on initial page load?  If the panel is open by default, but a user has not selected a Connector, what is on the panel? (for the Source, Destination, and Onboarding workflows)
- The workflow in the Connections page is unique -- a user might be selecting an existing Source/Destination or creating a new one.  How does the side panel fit in here?  (if it is open by default, what is the placeholder? What happens when a user picks an existing Connector? Or: redirect to a different page with the side panel open?  Side panel appears when user picks a Connector from the dropdown?)

## Describe the alternative you’ve considered or used
Currently, the side panel opens when a user clicks "Setup Guide" from within the Create a Source/Destination forms.

## Additional context
Though related, this doesn't block / is not blocked by https://github.com/airbytehq/airbyte/issues/11434 to link the form fields to different sections of the docs

**Comment by nataliekwong** [comment]
> Not all of our Sources and Destinations have internal docs (for example, Orb). How should we handle that if the panel is open by default?
> Some do not have external docs linked either.

@tealjulia Would this be the docs page for Orb? https://docs.airbyte.com/integrations/sources/orb
@Amruta-Ranade do you know of any connectors in Cloud where we don't have a docs page? This mismatch surprises me.

**Comment by nataliekwong** [comment]
@andyjih The questions from Teal should be addressed in the new design from Nico, but would appreciate your validation on the final approval for the design!

**Comment by tealjulia** [comment]
Thanks Natalie! The issue was with a place we were fetching the URL's from in the back end. I think there were only two connectors without docs after I found that. Updating the note above!

**Comment by andyjih** [comment]
> What should a user see on initial page load? If the panel is open by default, but a user has not selected a Connector, what is on the panel? (for the Source, Destination, and Onboarding workflows)

Would it be possible to expand the panel after the user has selected a Connector? 

> The workflow in the Connections page is unique -- a user might be selecting an existing Source/Destination or creating a new one. How does the side panel fit in here? (if it is open by default, what is the placeholder? What happens when a user picks an existing Connector? Or: redirect to a different page with the side panel open? Side panel appears when user picks a Connector from the dropdown?)

Same as above-- if possible, let's expand the side panel after the user has selected the Connector.

If expanding the side panel after selection is a can of worms, we can have it expanded by default and ask Nico for a placeholder image that we can show. 

How does that sound?

**Comment by Amruta-Ranade** [comment]
@Amruta-Ranade do you know of any connectors in Cloud where we don't have a docs page? This mismatch surprises me.

AFAIK, all Cloud connectors have docs. LMK if you find any missing!

**Comment by tealjulia** [comment]
> > What should a user see on initial page load? If the panel is open by default, but a user has not selected a Connector, what is on the panel? (for the Source, Destination, and Onboarding workflows)
> 
> Would it be possible to expand the panel after the user has selected a Connector?
> 
> > The workflow in the Connections page is unique -- a user might be selecting an existing Source/Destination or creating a new one. How does the side panel fit in here? (if it is open by default, what is the placeholder? What happens when a user picks an existing Connector? Or: redirect to a different page with the side panel open? Side panel appears when user picks a Connector from the dropdown?)
> 
> Same as above-- if possible, let's expand the side panel after the user has selected the Connector.
> 
> If expanding the side panel after selection is a can of worms, we can have it expanded by default and ask Nico for a placeholder image that we can show.
> 
> How does that sound?

That's absolutely possible.  I was concerned having it pop up when a user selects a connector might be a jarring user experience.  Would not be difficult to implement.

**Comment by tealjulia** [comment]
> @Amruta-Ranade do you know of any connectors in Cloud where we don't have a docs page? This mismatch surprises me.
> 
> AFAIK, all Cloud connectors have docs. LMK if you find any missing!

I was not able to find documentation for Google Firestore or Mailgun.  Is there a place I should submit an issue/ticket for those?

**Comment by Amruta-Ranade** [comment]
Gotcha. Can you [open a documentation issue](https://github.com/airbytehq/airbyte/issues/new?assignees=&labels=type%2Fenhancement%2C+area%2Fdocumentation&template=documentation.md&title=) and assign it to me?

**Comment by octavia-squidington-iii** [comment]
cc @airbytehq/frontend

**Comment by tealjulia** [comment]
I found one additional place that currently links to documentation which is the Settings pages for Sources and Destinations (the page where you view/edit an already existing Source/Destination).  Should the Setup Guides display by default on these also?  Open in a side panel via a link?

**Comment by nataliekwong** [comment]
No, I don't think we want the setup guide to open there (automatically or otherwise), since users aren't navigating to that part of the app to actively set something up or even troubleshoot a connector.

When people navigate to that page it's typically just to browse (like a catalog) of the available connectors in Cloud, rather than having to scroll through a dropdown.

**Comment by andyjih** [comment]
@nataliekwong Just to double-check, I _think_ it's worth still linking to the Setup Guide and showing that content in a side panel on the Edit Source page, even if we don't automatically open it. Are you suggesting we remove the link entirely from this page?
  
<img width="1728" alt="2022-05-17_11-25-47" src="https://user-images.githubusercontent.com/633226/168884310-a3f2945f-2747-428b-91c4-c1ef90da12ad.png">

**Comment by andyjih** [comment]
A-ha! Yes, agreed. :)

**Comment by tealjulia** [comment]
It's looking pretty good!  I think folks using a mobile device to configure their Cloud setup is probably an edge case... but wondering if we want to consider it here.  The two easiest solutions are:

No documentation panel, remains the same as before the side panel.  Some forms may be cut off but are no worse/different than before:
<img width="390" alt="Screen Shot 2022-05-24 at 1 33 46 PM" src="https://user-images.githubusercontent.com/63751206/170097481-ffdb1cb5-5d4a-4c64-a281-9bfeddd7b714.png">

Dark overlay (I have this displaying when the panel width is <450px... may not make sense on a phone where you _can't_ resize it large enough):
<img width="391" alt="Screen Shot 2022-05-24 at 1 32 51 PM" src="https://user-images.githubusercontent.com/63751206/170097585-2c816595-d01e-4961-a880-2feb3f9f3aea.png">

Neither involves having the docs panel open.  Open to other ideas.

**Comment by tealjulia** [comment]
@andyjih Any product input on how this feature should behave on mobile?  I know it's not currently a big priority.  Our options are: 

(1) Show the forms how they previously existed with no setup guide:
<img width="390" alt="Screen Shot 2022-05-24 at 1 33 46 PM" src="https://user-images.githubusercontent.com/63751206/170107050-d05c0536-13e5-4e66-891b-7b44d4519408.png">

(2) Show this page (probably different wording...)?
<img width="391" alt="Screen Shot 2022-05-24 at 1 32 51 PM" src="https://user-images.githubusercontent.com/63751206/170107001-c74bfee2-3528-43ee-91b2-e489a1b498da.png">

(3) Other?

Didn't want to spend more time on it than Product thinks it's worth.

**Comment by andyjih** [comment]
@tealjulia Option 1 looks good to me for now. Mobile is pretty low priority at the moment. Thanks!
